manifoldco/shadowcat
# Load DD and log duration between load time and token received

## Reason for change

This PR is to load the DataDog client and log duration from load -> token received event

DataDog doesn't have a RESTful way to submit logs from the frontend. We would need to expose our API key on the frontend to do that. Using the more secure client token requires use of their browser client. They don't seem to publish an NPM package of that client, so I'm loading via a script tag here. The caveat is that if shadowcat gets a Manifold token before DD loads, we don't end up submitting a log - I think this is fine since we're more concerned with logging slowness.

## Testing

1. Load dev server locally
1. Check DataDog logs for "Token received" event
1. Click that event and there should be a duration property
